### DISCUSSION TITLE
Uno will support native aot?

### DISCUSSION THREAD

**Discussion by wuliangtdi** [unoplatform/uno#8089] [Q&A]
.net 7 will support native aot,so the uno also will support native aot(not uwp.I know uwp support aot)

**Comment by robloo** [comment]
Native AOT compilation is supported using Mono on Android and iOS already. I'm not sure for macOS and the Skia backends what the plan is there.

**Reply by jeromelaban** [reply]
Native AOT is a .NET 7 feature, and Uno is using .NET 7 for Skia targets. This means that Native AOT will be available (and likely already is with current nightlies).

Native AOT is not supported on WebAssembly though, and is not used on iOS/Android/Catalyst and it's the AOT compilation which is used in these cases.

**Reply by robloo** [reply]
Ok, so native compilation is/soon-will-be supported on all platforms: just not everything is moving to NativeAOT with .NET 7 and some are still using older (but well tested/widely-used) mono AOT. It is a slow unification of .NET/Core/Mono but I'm sure Microsoft will unify all this in the future too.

**Comment by ivanjx** [comment]
@jeromelaban is there any guide or tutorial to enable this on skia backends currently? thanks

**Comment by ivanjx** [comment]
tried to build with native aot but encountered this error:
```
Unhandled Exception: System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Reflection.Runtime.Assemblies.RuntimeAssemblyInfo.GetReferencedAssemblies() + 0x20
   at GLib.GType.LookupType(IntPtr) + 0x282
   at GLib.ObjectManager.GetTypeOrParent(IntPtr) + 0x2d
   at GLib.ObjectManager.CreateObject(IntPtr) + 0x2c
   at GLib.Object.GetObject(IntPtr, Boolean) + 0x11e
   at Gdk.Screen.get_Default() + 0x2d
   at Uno.UI.Runtime.Skia.GtkHost.SetupTheme() + 0x33
   at Uno.UI.Runtime.Skia.GtkHost.Run() + 0x28
   at unotest.Skia!<BaseAddress>+0x1c5fe0b
Aborted (core dumped)
```

gtk csproj:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
    <OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <Nullable>enable</Nullable>
    <PublishAot>true</PublishAot>
    <StripSymbols>true</StripSymbols>
    <PublishTrimmed>true</PublishTrimmed>
  </PropertyGroup>
  <ItemGroup Condition="exists('..\unotest.Windows')">
    <EmbeddedResource Include="..\unotest.Windows\Package.appxmanifest" LogicalName="Package.appxmanifest" />
    <Content Include="..\unotest.Windows\Images\StoreLogo.png" Link="Assets\StoreLogo.png" />
  </ItemGroup>
  <ItemGroup>
    <Content Include="Assets\**" />
  </ItemGroup>
  <ItemGroup>
    <UpToDateCheckInput Include="..\unotest.Shared\**\*.xaml" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.WinUI.Skia.Gtk" Version="4.6.19" />
    <PackageReference Include="Uno.WinUI.RemoteControl" Version="4.6.19" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.6.19" />
    <PackageReference Include="Uno.WinUI.Lottie" Version="4.6.19" />
    <PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.3" />
    <PackageReference Include="SkiaSharp.Skottie" Version="2.88.3" />
  </ItemGroup>
  <Import Project="..\unotest.Shared\unotest.Shared.projitems" Label="Shared" />
</Project>
```

**Reply by ivanjx** [reply]
cmd: `dotnet publish -r linux-x64 -c Release`

**Reply by ivanjx** [reply]
sdks:
```
5.0.408 [/usr/share/dotnet/sdk]
6.0.302 [/usr/share/dotnet/sdk]
6.0.400 [/usr/share/dotnet/sdk]
7.0.100-preview.7.22377.5 [/usr/share/dotnet/sdk]
7.0.100-rc.1.22431.12 [/usr/share/dotnet/sdk]
7.0.100-rc.2.22477.23 [/usr/share/dotnet/sdk]
```

**Reply by jeromelaban** [reply]
_Make sure to continue your existing thread instead of opening a new one, it makes following content easier._

Yes, it's supported! Recent GtkSharp updates made this possible, see https://github.com/unoplatform/uno/pull/10351 on how to enable it.

**Reply by ivanjx** [reply]
thats great! is there any tutorial on how to try this now?

**Reply by jeromelaban** [reply]
Take a look at the PR, there's some new documentation.

**Reply by ivanjx** [reply]
is this PR currently included in any build of uno?

**Reply by jeromelaban** [reply]
The doc I added says the following:
_Building an Uno Platform Skia+GTK app with .NET (7+) Native AOT requires, GtkSharp 3.24.24.38 (or later), or Uno Platform 4.7 (or later)._

So you'll need net7, a version of Uno with that PR merged, or an explicit install of `GtkSharp` to `3.24.24.38` in your `.Skia.Gtk` project. Follow the rest of the steps and you should be good to go.

**Reply by ivanjx** [reply]
is there any dependencies for windows with skia gtk backend? i tried to build and works fine on linux but crashes on windows 10

**Reply by ivanjx** [reply]
nevermind i solved it by installing the gtk3 runtime. i think it would be better to package the runtime with the uno app though
